indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -305,6 +346,125 @@ public TaskLock lock(final Task task, final Interval interval) throws Interrupte
   }
 
   /**
+   * Check priorities of existing tasks, if all has lesser priorities and no lock is non-preemptive then return true
+   * If this task can acquire lock and some lower priority TaskLocks need to be revoked then populate tasksToPreempt Map with the entry - taskId and TaskLock
+   * */
+  private Pair<Boolean, Map<String, TaskLock>> canAcquireLock(final List<TaskLockPosse> foundPosses, int priority) {
+    boolean canAcquire = true;
+    final Map<String, TaskLock> tasksToPreempt = new HashMap<>();
+    for (TaskLockPosse taskLockPosse : foundPosses) {
+      if(priority <= taskLockPosse.getTaskLock().getPriority() || !taskLockPosse.getTaskLock().isPreemptive()) {
+        // Check whether the tasks (in the taskLockPosse) which are blocking the new task are actually running,
```

Comment:
removing it would mean this method doesn't need to make any metadata calls, which I think would be beneficial.